cpp/src/io/parquet/page_data.cu

```diff
@@ -1741,21 +1743,21 @@ struct start_offset_output_iterator {
 
   start_offset_output_iterator operator+ __host__ __device__(int i)
   {
-    return start_offset_output_iterator{p + i, src_col_schema, nesting_depth};
+    return start_offset_output_iterator{
+      pages, page_indices, cur_index + i, src_col_schema, nesting_depth};
   }
 
-  void operator++ __host__ __device__() { p++; }
+  void operator++ __host__ __device__() { cur_index++; }
 
-  reference operator[] __device__(int i) { return dereference(p + i); }
-  reference operator*__device__() { return dereference(p); }
+  reference operator[] __device__(int i) { return dereference(cur_index + i); }
+  reference operator*__device__() { return dereference(cur_index); }
 
  private:
-  reference __device__ dereference(PageInfo *p)
+  reference __device__ dereference(int index)
   {
-    if (p->src_col_schema != src_col_schema || p->flags & PAGEINFO_FLAGS_DICTIONARY) {
-      return empty;
-    }
-    return p->nesting[nesting_depth].page_start_value;
+    PageInfo const &p = pages[page_indices[index]];
+    if (p.src_col_schema != src_col_schema || p.flags & PAGEINFO_FLAGS_DICTIONARY) { return empty; }
```

Comment:
No.  It's basically a blank location for values that we don't want to add to the scan result.  I figured it would be better to output nothing for dictionary pages rather than write values that are meaningless to the actual struct values.